### PR TITLE
Tours: fix error on DQ

### PR DESCRIPTION
**Pull request by whalemer**
setMatchResult is already called when the battle is won through the forfeit

### CODE REVIEW

**Comment by whalemer** [comment]
oops, this doesn't remove everything, give me a second to make sure we can remove the rest.

**Comment by whalemer** [comment]
This is pretty much the cleanest solution I could find. Ideally I think you'd want to rework the node state system, but this works well enough imo.

**Comment by panpawn** [comment]
@sirDonovan ?

File: tournaments/index.js [hunk 1]
```diff
@@ -873,13 +873,14 @@ class Tournament {
 			this.update();
 			return this.room.update();
 		}
-
-		let error = this.generator.setMatchResult([from, to], result, score);
-		if (error) {
-			// Should never happen
-			return this.room.add("Unexpected " + error + " from setMatchResult([" + room.p1.userid + ", " + room.p2.userid + "], " + result + ", " + score + ") in onBattleWin(" + room.id + ", " + winnerid + "). Please report this to an admin.").update();
+		if (!(this.generator.users.get(from).isDisqualified || this.generator.users.get(to).isDisqualified)) {
+			// If a player was disqualified, handle the results there
```
Comment (Zarel) [inline tournaments/index.js]: Weren't we going to remove this from the "disqualified" code, rather than here?

**Comment by whalemer** [comment]
From what I can tell, this is happening because when you get DQd, two things happen:

-the DQ method sets a loss result for all "active" battles
-you forfeit your battles, which will eventually set the result for

The problem arises when the dq method happens before the battle is ended. Since the DQ method will have marked the battle as complete, after the battle actually ends there's nothing to update. 

You can't strictly remove the part in the dq method, because then it won't mark battles that are active but haven't started yet, and you can't remove the one when a battle is won, because that breaks normal wins. 

IMO the best solution would be to make the DQ method only mark battles as ended if they haven't started, but I don't see a way to do that that isn't terrible.

**Comment by Zarel** [comment]
You could change the part in the dq method to only mark the ones that aren't started?

**Comment by whalemer** [comment]
How? The generator can't tell the status of a match

**Comment by Zarel** [comment]
...I guess hacks are fine...

**Pull request merged**